### PR TITLE
Update Google FHIR SDK dependencies

### PR DESCRIPTION
**Pull request by allan-on**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Updated the Engine, SDC, Workflow and Contrib:Barcode dependencies to the latest released versions

See https://github.com/google/android-fhir/compare/master...opensrp:android-fhir:master-release

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [x] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt:61]: can we remove these lines? if it's ok to comment them, why are they needed?

**Review by pld — CHANGES_REQUESTED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt
Comment (ellykits) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt:61]: I believe that can be addressed separately

Comment (pld) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt:61]: OK, but if this means we will not be able to evaluate measures, we can't merge this without the fix

Comment (ellykits) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt:61]: Yea. @allan-on can you tests this? I will also tag @maimoonak.

Comment (ellykits) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt:61]: Yea. @allan-on can you tests this? I will also tag @maimoonak.

Comment (pld) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt:61]: K thanks, and if we still can evaluate measures, then I think we can remove these lines until we need to use `loadLib` and address them then

Comment (allan-on) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt:61]: Removing these lines will break evaluation of measures. 

In summary, the changes on the SDK introduce loading of Resources, for instance, a Library that is used for evaluation of Immunizations, from the [KnowledgeManager](knowledge/src/main/java/com/google/android/fhir/knowledge/KnowledgeManager.kt). The KnowledgeManager persists them in a separate database and when we do evaluate any Measure there's a call to  the `knowledgeManager.loadResources()`. 

So in this specific implementation, the `loadLib()` was removed and instead there's [install(implementationGuide: ImplementationGuide, rootDirectory: File)](https://github.com/google/android-fhir/blob/master/knowledge/src/main/java/com/google/android/fhir/knowledge/KnowledgeManager.kt#L58 )
or [install(file: File)](https://github.com/google/android-fhir/blob/master/knowledge/src/main/java/com/google/android/fhir/knowledge/KnowledgeManager.kt#L76)

The refactors required on our end would probably need to make use of the provided implementations. 

From my understanding the resources required will already be available in the DB but in a different `ResourceEntity` table. 

cc @ellykits @pld @maimoonak

Comment (pld) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt:61]: OK, that's reasonable, and means we need to resolve this TODO before we can merge this PR

Comment (allan-on) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt:61]: The other alternative to get the rest of the changes in faster would be to hold off on updating the `workflow` dependency version, and then draft an implementation plan for using the `KnowledgeManager` for Knowledge Resources essentially packaged up in an Implementation Guide such as ValueSets, Plan/Activity Definitions and Library Resources containing CQL.

**Comment by pld** [comment]
That approach makes sense to me

**Comment by f-odhiambo** [comment]
Passed QA for sync failure at 100%

File: android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt
Comment (pld) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/FhirEngineExtension.kt:61]: Let's do that so we can get the other updates merged, then have a separate PR to update workflow and integrate that w/measure eval

**Review by ndegwamartin — APPROVED** [review]

**Pull request merged**
